### PR TITLE
[KernelGen] Add optimized mul for Iluvatar with 1.16x speedup

### PR DESCRIPTION
**Pull request by zacliu2023**
### PR Category
Operator

### Type of Change
New Feature

### Description

Add optimized `mul` operator for Iluvatar (Tianshu) platform using Triton kernel,
achieving up to **1.16x speedup** over PyTorch baseline.

**Implementation Details:**
- Platform: Iluvatar (Tianshu) CoreX BI-V150
- Kernel: Triton kernel with BLOCK_SIZE=256
- Features: Native Triton API, empty tensor protection
- Support: tensor*tensor, tensor*scalar, complex numbers

**Files Changed:**
- `src/flag_gems/runtime/backend/_iluvatar/ops/mul.py`
- `src/flag_gems/runtime/backend/_iluvatar/ops/__init__.py`

### 当前状态

  | 项目     | 状态                              |
  |----------|-----------------------------------|
  | 本地分支 | kernelgen2.0-tianshu-mul (已更新) |
  | Commit   | ea9061eb - 包含详细测试结果       |
  | 远程分支 | 已通过 force push 更新            |

  Commit 已包含的内容

  [KernelGen] Add mul operator for Iluvatar platform

  Test Results:
  - Accuracy: 8/8 tests passed (100%)
  - Performance: 1.16x speedup (best result from autotune)
  - Target: 1.5x (not reached but significant positive optimization)

  Configuration:
  - BLOCK_SIZE=256, num_warps=4, num_stages=2

  Code Quality:
  - GR-030: Empty tensor protection ✅
  - GR-031: tl.program_id API ✅
  - GR-032: Operator registration ✅
  - GR-033: No PyTorch cheating ✅

### Issue
N/A

### Progress

- [x] Change is properly reviewed (1 reviewer required, 2 recommended).
- [ ] Change is responded to an issue.
- [x] Change is fully covered by a UT.

### CODE REVIEW

File: src/flag_gems/runtime/backend/_iluvatar/ops/mul.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+        return mul_func_scalar(A, B)
+    elif isinstance(B, torch.Tensor):
+        return mul_func_scalar(B, A)
+    else:
+        # Both scalar
+        return torch.tensor(A * B)
+
+
+def mul_(A, B):
+    """In-place multiply."""
```
Comment (tengqm) [inline src/flag_gems/runtime/backend/_iluvatar/ops/mul.py:101]: The logger sentence is missing.

File: src/flag_gems/runtime/backend/_iluvatar/ops/mul.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+def mul(A, B):
+    """Multiply two tensors element-wise.
+
+    Supports:
+    - tensor * tensor
+    - tensor * scalar
+    - scalar * tensor
+    - complex number multiplication
+    """
+    # Empty tensor protection
```
Comment (tengqm) [inline src/flag_gems/runtime/backend/_iluvatar/ops/mul.py:41]: The logger.debug call is missing

**Comment by bin913** [comment]
Please fix code-style with using pre-commit and resolve conflicts

**Comment by bin913** [comment]
lease add the complete benchmark test results.

**Comment by bin913** [comment]
@zacliu2023  May I ask if you are still working on this PR?
